File: app/src/main/java/org/cuberite/android/receivers/StartupReceiver.java
package org.cuberite.android.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;

import org.cuberite.android.helpers.CuberiteHelper;
import org.cuberite.android.helpers.StateHelper;

import static android.content.Context.MODE_PRIVATE;

public class StartupReceiver extends BroadcastReceiver {
    public void onReceive(Context context, Intent intent) {
        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            final SharedPreferences preferences = context.getSharedPreferences(context.getPackageName(), MODE_PRIVATE);

            if (preferences.getBoolean("startOnBoot", false)
                    && StateHelper.isCuberiteInstalled(context)) {
                CuberiteHelper.startCuberite(context);
            }
        }
    }
}

File: app/src/main/kotlin/org/cuberite/android/receivers/StartupReceiver.kt
package org.cuberite.android.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import org.cuberite.android.MainApplication
import org.cuberite.android.services.CuberiteService
import org.cuberite.android.services.InstallService

class StartupReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (Intent.ACTION_BOOT_COMPLETED == intent.action) {
            if (MainApplication.preferences.getBoolean("startOnBoot", false)
                    && InstallService.isInstalled) {
                CuberiteService.start(context)
            }
        }
    }
}
